src/server/index.js

```diff
@@ -43,6 +62,37 @@ app.get(['/', '/login', '/register'], (req, res) => {
   res.sendFile(path.resolve(__dirname, '../../dist/index.html'));
 });
 
+app.post('/api/register', jsonParser, function(req, res) {
+  if (req.headers['content-type'] !== 'application/json') {
+    res.json(contentTypeError);
+    return;
+  }
+  if (!req.body.username) {
+    res.status(400).send(userNameMissing);
+    return;
+  }
+  if (!req.body.password) {
+    res.json(passWordMissing);
+    return;
+  }
+
+  req.body.password = generateHash(req.body.password);
```

Comment:
do not override any property of the `req.body` you might need it later. create a new variable like `const passwordHash`